guides/source/active_support_instrumentation.md

```diff
@@ -3,623 +3,1144 @@
+```ruby
+sql_counter = 0
+ActiveSupport::Notifications.subscribe "sql.active_record" do |event|
+   sql_counter += 1
+end
+```
+
+Similarly, when a controller action finishes processing, it instruments the
+`process_action.action_controller` event. This event includes data such as the
+controller name, action name, and request information in its payload.
 
 ```ruby
 ActiveSupport::Notifications.subscribe "process_action.action_controller" do |event|
-  event.name        # => "process_action.action_controller"
-  event.duration    # => 10 (in milliseconds)
-  event.allocations # => 1826
-  event.payload     # => {:extra=>information}
+   Rails.logger.info "Duration: #{event.duration}"
+end
+```
+
+You can read more about creating your own events and subscribing to events in
+the [Instrumenting Events section](#instrumenting-events) and [Subscribing to an
+Event section](#subscribing-to-an-event) respectively.
+
+### Subscribers
+
+A subscriber is the object that is used to subscribe to events. It is used to
+listen to events and perform some action when an event is emitted.
+
+For example, if you want to subscribe to the `process_action.action_controller`
+event and benchmark it, then you can create a subscriber that listens to that
+event. The subscriber will then be able to perform other actions when the event
+is emitted, such as logging the duration of the action.
+
+A single event can have multiple subscribers. You can read more about
+subscribers in the [Subscribing to an Event section](#subscribing-to-an-event).
+
+Instrumenting Events
+--------------------
 
-  Rails.logger.info "#{event} Received!"
+Rails provides built-in events which you can read more about in the [Rails
+Framework Instrumentation section](#rails-framework-instrumentation). However,
+there may be instances where you might want to instrument your own event.
+
+To instrument a custom event, you can call
+[`ActiveSupport::Notifications.instrument`](https://api.rubyonrails.org/classes/ActiveSupport/Notifications.html#method-c-instrument)
+with the `name` of your custom event, a `payload` which is a hash containing
+information about the event, and an optional block.
+
+```ruby
+ActiveSupport::Notifications.instrument "publish.posts", { title: "My Post", author: "John Doe" } do
+  # Publish the post here
 end
 ```
 
-If you don't need all the data recorded by an Event object, you can also specify a
-block that takes the following five arguments:
+TIP: When defining your own event names, follow Rails conventions. The
+recommended format used for [Rails Framework
+Instrumentation](#rails-framework-instrumentation) is: `<action>.<component>`,
+where `<action>` describes what happened (e.g., `service_delete`,
+`start_transaction`) and `<component>` is the framework or library name (e.g.,
+`active_record`, `active_storage`). Hence, if you want to instrument an event
+when a post is published in a blogging application, you could use the event name
+`publish.posts`.
```

Comment:
There was some discussion about naming for Event Reporter over here:
https://github.com/rails/rails/pull/56156#discussion_r2539287558

I think idea is that you _can_ define some semantics for naming, like `<action>.<component>` but I don't think we need to suggest it as preferred? I like using this part to explain the **Rails** convention for event naming throuought the framework